### PR TITLE
SL-503: Fix data table suggestions in App Lab text mode

### PR DESCRIPTION
**Pull request by ebeastlake**
The following PR fixes the bug described in https://codedotorg.atlassian.net/browse/SL-503. 

## Links

Jira ticket: https://codedotorg.atlassian.net/browse/SL-503

## Testing story

New sanitization logic was added to a function called `stripEncapsulatingDoubleQuotes` in `src/utils.js`. Unit tests were added for this function. 

The tests in `apps/test/unit/storage/getColumnsDropdownTest.js` were not modified because 1) there is a 2-hour timebox on this task and 2) the existing tests are testing a lot more than they should be; namely, the highly nested code paths used for getting param values in both block and text modes that are mostly unrelated to my change.

I also tested that the new logic works manually by testing the choosing of a column name in both block and text mode (video below). 

https://user-images.githubusercontent.com/26844240/219503469-eace7367-d956-4249-8fb8-cd11116b676c.mov

## Follow-up work

Researching this task revealed a more nuanced bug with the `formatParamString` function that I captured here: https://codedotorg.atlassian.net/browse/SL-580

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/dropletUtils.js [hunk 1]
```diff
@@ -1147,7 +1147,9 @@ export function setParamAtIndex(index, value, block) {
 
 /**
  * Take a string like "'param1', 'param2'" and an index and return
- * the param at the given index without extra quotes, commas or spaces.
+ * the param at the given index without extra quotes, commas or spaces
+ * Note: The sanitization here is too aggressive for some parameters and
+ * is the cause of at least one bug: https://codedotorg.atlassian.net/browse/SL-580
```
Comment (fisher-alice) [inline apps/src/dropletUtils.js:1152]: Nice sleuthing on this!

File: apps/test/unit/utilsTest.js [hunk 2]
```diff
@@ -772,4 +773,29 @@ describe('utils modules', () => {
...
+  describe('stripEncapsulatingDoubleQuotes', () => {
+    let result;
+
+    it('removes double quotes at the first and last character position', () => {
+      result = stripEncapsulatingDoubleQuotes('"blah"');
+      expect(result).to.equal('blah');
+
+      result = stripEncapsulatingDoubleQuotes('" "');
+      expect(result).to.equal(' ');
+    });
```
Comment (fisher-alice) [inline apps/test/unit/utilsTest.js:789]: Could add a unit test for string `'""blah""'` -> `'"blah"'`

**Review by fisher-alice — APPROVED** [review]
LGTM!

File: apps/src/storage/getColumnDropdown.js [hunk 3]
```diff
@@ -39,10 +40,10 @@ export function getTables() {
 
 function getTableNameFromColumnSocket(socket, editor) {
   const paramValue = getFirstParam('getColumn', socket.parent, editor);
-  // The socket value has an extra set of double quotes. Trim off the first
-  // and last characters to remove, but don't use utils.stripQuotes because
-  // there may be other quotes in the table name (for example, apostrophes)
-  return paramValue.substring(1, paramValue.length - 1);
+
+  // The socket value (ex. the table name) has an extra set of double quotes in Droplet mode but not in text mode
+  // N.B. formatParamString is what removes the double quotes before we get here in text mode
```
Comment (molly-moen) [inline apps/src/storage/getColumnDropdown.js]: what does N.B. stand for?

**Review by molly-moen — APPROVED** [review]
looks good! just one acronym question

File: apps/src/storage/getColumnDropdown.js (hunk 3 above)
Comment (ebeastlake) [reply apps/src/storage/getColumnDropdown.js]: Here's the Wikipedia article: https://en.wikipedia.org/wiki/Nota_bene. This is something I learned in math classes in high school and college and have always taken to mean "tricky sidenote"/"don't forget." I thought was not that niche, but it looks like it is pretty niche and I'll just change it to `Note:` 😂

**Pull request merged**